### ISSUE TITLE
Query gnereration only happen with test sets n QueryGenerator class

### ISSUE THREAD

**Issue by sshivam95** [dice-group/dice-embeddings#281]
In the method [`generate_queries`](https://github.com/dice-group/dice-embeddings/blob/94ab3056aee494020e9b0090f1707aa3690d5e91/dicee/query_generator.py#L444), the mode should decide which edges should be passed to ground queries for getting queries and answers (as per the docstring). However, only test set is used. There is no functionality to use just the train or validation set.
